src/hiscores.ts

```diff
@@ -280,91 +285,81 @@ export async function getStats(
   ];
   const shouldGetFormattedRsn = options?.shouldGetFormattedRsn ?? true;
 
-  const mainRes = await httpGet<HiscoresResponse>(
-    getStatsURL('main', rsn, true),
-    options?.axiosConfigs?.main
-  );
-  if (mainRes.status === 200) {
-    const emptyResponse: AxiosResponse<HiscoresResponse> = {
-      status: 404,
-      data: { skills: [], activities: [] },
-      statusText: '',
-      headers: {},
-      config: {}
-    };
-    const getModeStats = async (
-      mode: Extract<Gamemode, 'ironman' | 'hardcore' | 'ultimate'>
-    ): Promise<AxiosResponse<HiscoresResponse>> =>
-      otherGamemodes.includes(mode)
-        ? httpGet<HiscoresResponse>(
-            getStatsURL(mode, rsn, true),
-            options?.axiosConfigs?.[mode]
-          ).catch((err) => err)
-        : emptyResponse;
-    const formattedName = shouldGetFormattedRsn
-      ? await getRSNFormat(rsn, options?.axiosConfigs?.rsn).catch(
-          () => undefined
-        )
-      : undefined;
-
-    const player: Player = {
-      name: formattedName ?? rsn,
-      mode: 'main',
-      dead: false,
-      deulted: false,
-      deironed: false
-    };
-    player.main = parseJsonStats(mainRes.data);
-
-    const ironRes = await getModeStats('ironman');
-    if (ironRes.status === 200) {
-      player.ironman = parseJsonStats(ironRes.data);
-      const hcRes = await getModeStats('hardcore');
-      const ultRes = await getModeStats('ultimate');
-      if (hcRes.status === 200) {
-        player.mode = 'hardcore';
-        player.hardcore = parseJsonStats(hcRes.data);
-        if (
-          player.ironman.skills.overall.xp !== player.hardcore.skills.overall.xp
-        ) {
-          player.dead = true;
-          player.mode = 'ironman';
-        }
-        if (
-          player.main.skills.overall.xp !== player.ironman.skills.overall.xp
-        ) {
-          player.deironed = true;
-          player.mode = 'main';
-        }
-      } else if (ultRes.status === 200) {
-        player.mode = 'ultimate';
-        player.ultimate = parseJsonStats(ultRes.data);
-        if (
-          player.ironman.skills.overall.xp !== player.ultimate.skills.overall.xp
-        ) {
-          player.deulted = true;
-          player.mode = 'ironman';
-        }
-        if (
-          player.main.skills.overall.xp !== player.ironman.skills.overall.xp
-        ) {
-          player.deironed = true;
-          player.mode = 'main';
-        }
-      } else {
+  const main = await getOfficialStats(rsn, 'main', options?.axiosConfigs?.main);
+
+  const emptyResponse: HiscoresResponse = {
+    skills: [],
+    activities: []
+  };
+  const getModeStats = async (
+    mode: Extract<Gamemode, 'ironman' | 'hardcore' | 'ultimate'>
+  ): Promise<HiscoresResponse | undefined> =>
+    otherGamemodes.includes(mode)
+      ? getOfficialStats(rsn, mode, options?.axiosConfigs?.[mode])
+      .catch(err => err)
+      : emptyResponse;
```

Comment:
Now that this no longer returns an `AxiosResponse` object, and we are no longer checking for status code `200` below, I think we have to return `undefined` instead of `emptyResponse` here.

It looks like these changes as-is will modify behavior for gamemodes that are filtered out, and instead of omitting the excluded gamemodes, they will be present with `-1` for all ranks and scores.

Probably worth adding a test case to ensure that excluded gamemodes are omitted in the returned `Player` object.